tst/EnergyPlus/unit/IceThermalStorage.unit.cc

```diff
@@ -0,0 +1,180 @@
+//
+// (4) Use of EnergyPlus(TM) Name. If Licensee (i) distributes the software in stand-alone form
+//     without changes from the version obtained under this License, or (ii) Licensee makes a
+//     reference solely to the software portion of its product, Licensee must refer to the
+//     software as "EnergyPlus version X" software, where "X" is the version number Licensee
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+// EnergyPlus::Low Temperature Radiant Unit Tests
+
+// Google Test Headers
+#include <gtest/gtest.h>
+
+// EnergyPlus Headers
+#include "Fixtures/EnergyPlusFixture.hh"
+#include <EnergyPlus/IceThermalStorage.hh>
+#include <EnergyPlus/CurveManager.hh>
+#include <EnergyPlus/UtilityRoutines.hh>
+
+#include <EnergyPlus/General.hh>
+
+using namespace EnergyPlus;
+using namespace EnergyPlus::IceThermalStorage;
+using namespace ObjexxFCL;
+using namespace DataGlobals;
+using namespace CurveManager;
+using namespace EnergyPlus::General;
+
+class IceThermalStorageTest : public EnergyPlusFixture
+{
+public:
+    int TotDetailedIce;
+
+protected:
+    virtual void SetUp()
+    {
+        EnergyPlusFixture::SetUp(); // Sets up the base fixture first.
+
+    }
+
+    virtual void TearDown()
+    {
+        EnergyPlusFixture::TearDown(); // Remember to tear down the base fixture after cleaning up derived fixture!
+    }
+};
+
+TEST_F(EnergyPlusFixture, IceThermalStorage_CalcQstarTest)
+{
+    int TotDetailedIce = 4;
+    int TotCurves = 4;
+    int IceStorageCurveType;
+    int TestNum;
+    Real64 CurveAnswer = 0.0;
+    Real64 ExpectedValue = 0.0;
+    Real64 Tolerance = 0.001;
+    
+    if (allocated(IceThermalStorage::DetIceStor)) DetIceStor.deallocate();
+    if (allocated(CurveManager::PerfCurve)) PerfCurve.deallocate();
```

Comment:
These should be deallocated in a `clear_state` function. Oh, I see, IceThermalStorage doesn't have one. Follow the example of, say Pumps.cc. Every module-level variable should be reset to it's initial value and every array deallocated in a `clear_state` function, and then the new function needs to get added to [`EnergyPlusFixture::clear_all_states()`](https://github.com/NREL/EnergyPlus/blob/af8b6e160fa36a0cc5d5eaa28db46883e97a2e96/tst/EnergyPlus/unit/Fixtures/EnergyPlusFixture.cc).

Reply:
@mjwitte Ok, IceThermalStorage now has a clear_state and the checks noted above have been deleted.